From a8cc53df6d859ce0794f8489aa1ab3b6f6051a97 Mon Sep 17 00:00:00 2001
From: "Brint E. Kriebel" <bekit@cyngn.com>
Date: Wed, 14 Feb 2018 23:02:06 -0800
Subject: [PATCH 44/90] releasetools: Use the first entry of a mount point when
 reading fstab

There may be multiple entries in fstab for a mount point. Use the first
value found so the entries are prioritized in order.

Change-Id: Ibd2631413d3e00507d3a0ec84fb91755f1c7993c
Ticket: OPO-326
---
 tools/releasetools/common.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index b3691d2f3..3d1018425 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -340,8 +340,9 @@ def LoadRecoveryFSTab(read_helper, fstab_version, recovery_fstab_path,
         context = i
 
     mount_point = pieces[1]
-    d[mount_point] = Partition(mount_point=mount_point, fs_type=pieces[2],
-                               device=pieces[0], length=length, context=context)
+    if not d.get(mount_point):
+        d[mount_point] = Partition(mount_point=mount_point, fs_type=pieces[2],
+                                   device=pieces[0], length=length, context=context)
 
   # / is used for the system mount point when the root directory is included in
   # system. Other areas assume system is always at "/system" so point /system
-- 
2.17.0

